include/geometric_shapes/check_isometry.h

```diff
@@ -0,0 +1,143 @@
+ */
+namespace bodies
+{
+#ifndef CHECK_ISOMETRY_PRECISION
+/** \brief The default precision to which the transform has to correspond to an isometry. */
+#define CHECK_ISOMETRY_PRECISION Eigen::NumTraits<double>::dummy_precision()
+#endif
+
+namespace
+{
+/**
+ * \brief Return the number of decimal digits that should be printed to print all digits important for the given float
+ *        precision. E.g. when floatPrecision is 1e-12, printPrecision is 12 + 1 = 13.
+ * \param floatPrecision The float precision to which float numbers are considered.
+ * \return The desired number of decimal digits (after decimal dot). The result will always be at least 6.
+ * \note Used as a parameter for std::setprecision().
+ */
+inline int getPrintPrecision(const double floatPrecision)
+{
+  return std::max(6, static_cast<int>(std::abs(std::log10(floatPrecision))) + 1);
+}
+}  // namespace
+
+/**
+ * \brief Check whether the given transform is really (mathematically) an isometry.
+ * \param transform The possibly non-isometric transform.
+ * \param precision Precision to which the transform has to correspond to an isometry (element-wise).
+ * \param printError Whether an error should be printed to std::cerr with details about the false isometry.
+ * \return Whether the given transform is close to an isometry or not.
+ */
+inline bool checkIsometry(const Eigen::Isometry3d& transform, const double precision = CHECK_ISOMETRY_PRECISION,
+                          const bool printError = true)
+{
+  if (!transform.matrix().row(3).isApprox(Eigen::Vector4d::UnitW().transpose(), precision))
+  {
+    if (printError)
+    {
+      std::cerr << "The given transform is not an isometry! Its last row is: " << std::endl
+                << std::setprecision(getPrintPrecision(precision)) << std::fixed << transform.matrix().row(3)
+                << std::endl;
+    }
+    return false;
+  }
+
+  Eigen::Isometry3d::LinearMatrixType scale;
+  transform.computeRotationScaling((Eigen::Isometry3d::LinearMatrixType*)nullptr, &scale);
+  if (!scale.isApprox(Eigen::Matrix3d::Identity(), precision))
+  {
+    if (printError)
+    {
+      std::cerr << "The given transform is not an isometry! Its linear part involves non-unit scaling: " << std::endl
+                << std::setprecision(getPrintPrecision(precision)) << std::fixed << scale.diagonal().transpose()
+                << std::endl;
+    }
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * \brief Call a failed assertion if the given transform is not an isometry.
+ * \param transform The possibly non-isometric transform.
+ * \param precision Precision to which the transform has to correspond to an isometry (element-wise).
+ */
+inline void assertIsometry(const Eigen::Isometry3d& transform, const double precision = CHECK_ISOMETRY_PRECISION)
+{
+  if (!::bodies::checkIsometry(transform, precision))
+    assert(!"Invalid isometry transform");
+}
+
```

Comment:
I suggest avoiding a function here, because file information will not resolve to the correct line number in that case.
I have to admit that the previous solution didn't correctly resolve as well (due to my suggestion).

Reply:
So you suggest the `checkIsometry` should be a macro instead?

Reply:
No, checkIsometry should stay a function as originally suggested by me.
But, the content of `assertIsometry` should be moved into the macro `ASSERT_ISOMETRY`.
Essentially, it is important to have all `assert` statements within a macro to correctly resolve the line number.

Reply:
Fixed.